addons/mrp/migrations/12.0.2.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,120 @@
+---Fields in module 'mrp'---
+mrp          / mrp.bom                  / message_last_post (datetime)  : DEL 
+mrp          / mrp.bom                  / message_main_attachment_id (many2one): NEW relation: ir.attachment
+mrp          / mrp.bom                  / product_uom_id (many2one)     : relation is now 'uom.uom' ('product.uom')
+mrp          / mrp.bom.line             / product_uom_id (many2one)     : relation is now 'uom.uom' ('product.uom')
+# NOTHING TO DO
+
+mrp          / mrp.production           / activity_date_deadline (date) : not related anymore
+mrp          / mrp.production           / activity_date_deadline (date) : not stored anymore
```

Comment:
I'm seeing now that this is a problem in all models that implements `mail.activity.mixin`: if we keep the column in DB, there is going to be incorrect readings of it, so we must rename it. There can be 2 approaches:

* Rename the column on each module.
* Try to get all the existing fields on `mail` migration and rename in "batch".

Reply:
and which of the to approaches you recommend?

Reply:
If feasible, approach 2 will be desirable for doing things once, and not worrying on each module.

Reply:
but approach 2 I think is unfeasible. Because, in pre-migration of mail, there are a lot of modules that have not yet loaded.

Reply:
One possible thing is to look for any column with this name, and rename it. If one is incorrectly renamed, it can be re-renamed again on the module migration itself (in core I don't think it's needed).

Reply:
and how you get a list of all tables? 🤔

Reply:
maybe what should only do is just in the end-migration do a recompute of this column for all the models inheriting mail.activity.mixin.

Reply:
You can query certain special tables. Look how openupgradelib `column_exists` is implemented. And about end-migration, it doesn't work. The problem is the existence of the column, not any computation.